lib/dependencies/WorkerDependency.js

```diff
@@ -71,16 +73,34 @@ WorkerDependency.Template = class WorkerDependencyTemplate extends (
 		const chunk = entrypoint.getEntrypointChunk();
 
 		runtimeRequirements.add(RuntimeGlobals.publicPath);
-		runtimeRequirements.add(RuntimeGlobals.baseURI);
 		runtimeRequirements.add(RuntimeGlobals.getChunkScriptFilename);
 
-		source.replace(
-			dep.range[0],
-			dep.range[1] - 1,
-			`/* worker import */ ${RuntimeGlobals.publicPath} + ${
-				RuntimeGlobals.getChunkScriptFilename
-			}(${JSON.stringify(chunk.id)}), ${RuntimeGlobals.baseURI}`
-		);
+		const workerUrlTemplate = `${RuntimeGlobals.publicPath} + ${
+			RuntimeGlobals.getChunkScriptFilename
+		}(${JSON.stringify(chunk.id)})`;
+
+		if (dep.crossOriginLoading) {
+			runtimeRequirements.add(RuntimeGlobals.scriptUrl);
+			runtimeRequirements.add(RuntimeGlobals.createScriptBlob);
+
+			const baseUriTemplate = `JSON.stringify(${RuntimeGlobals.scriptUrl})`;
+			const scriptUrlTemplate = `JSON.stringify(new URL(${workerUrlTemplate}, ${RuntimeGlobals.scriptUrl}))`;
+			const crossOriginLoaderTemplate = `"__webpack_base_uri__=" + ${baseUriTemplate} + ";importScripts(" + ${scriptUrlTemplate} + ");"`;
```

Comment:
Bad idea to use `importScripts`, also it should respect ESM output

Reply:
Good point, I will create test cases for esm output and update the code :)

Reply:
I added support for ESM modules and tested it manually. Unfortunately, I wasn't able to make automatic tests to work with ESM (and I didn't found any ESM tests with workers) :/